### ISSUE TITLE
[FEATURE REQUEST] Ability to choose between PIN or password

### ISSUE THREAD

**Issue by victorgsfigueiredo** [muun/apollo#43]
Although Muun has a 3-attempts-policy for PIN unlocking, a 4-digit PIN has really low entropy. It would be nice instead of PIN to have a  password to access the app. This way, if my phone is stolen, the chances of getting my bitcoin stolen as well will be lower.
I already know Muun staff has plans to implement a lock feature for stolen devices, but it would be also nice to have a second layer of security too by having a strong password.

**Comment by champo** [comment]
Hi! We were thinking about raising the number of digits in the PIN from 4 to 6, would this be an improvement for you?

**Comment by victorgsfigueiredo** [comment]
Yes, it is a good improvement, but I would still choose to have a password if I had the option to. Maybe you can add this as a future improvement :-)

**Comment by EZ0010** [comment]
No please. Dont bring this to iOS. I find adding a password in any app in iOS a stupid idea and bad for UX and we all know muun wallet takes UX/UI in mind unlink many other wallets.
If the device itself has a password this is more than enough security. No one can access anything in the device. Adding an additional password would be silly idea and waste my time unlocking it.

If I have the chance I would even completely remove the PIN code from muun wallet. iOS security is more than enough. Maybe this would be needed on Android devices? idk

**Comment by victorgsfigueiredo** [comment]
@EZ0010 the request is to have the **option** to have a password, not to **enforce** it to all users. You would still be able to use a weak PIN if you'd like to.

**Comment by EZ0010** [comment]
> @EZ0010 the request is to have the **option** to have a password, not to **enforce** it to all users. You would still be able to use a weak PIN if you'd like to. 

If I have the chance I would even remove the **”weak”** PIN completely, iOS PIN to unlock the whole device is more than enough security for me (and 90% of people) and it is not possible to crack unless you are NSO.
Adding an additional PIN is just wasting time and making UX worst especially if it is required every 10 seconds (as muun do it).
Not to mention making that PIN (which is required every 10 seconds) a password that is more than 4 digits.

ofc it can be optional (for people who like pain and don't care about UX) but please don't force users to do it.

**Comment by victorgsfigueiredo** [comment]
Dude, nobody is forcing anyone here. I am simply suggesting a new optional feature for those who prefer security instead of usability.

**Comment by EZ0010** [comment]
You are free to submit your suggestion and I'm free to provide a feedback for it. Dude.

**Comment by victorgsfigueiredo** [comment]
Also, this suggestion is for muun/apollo, which is the repository for Android. Nobody is talking about iOS here.

**Comment by EZ0010** [comment]
When muun implement a new feature it is deployed to both platforms so providing feedback here for iOS also count.

**Comment by EZ0010** [comment]
I agree with your points and here are my thoughts regarding security while still having a good UX:

- Making the PIN code or biometrics authentication (or a along password for people who like pain) mandatory on the last screen of **sending** funds but not on receiving/viewing the wallet transactions.
- Adding an additional 2FA (with OTP on email if user opted in) when the user is trying to send more than X amount of money (say 500$) in one day.
